src/test/java/com/github/martapietka/ppm2pbmSpring/Ppm2PbmControllerTest.java

```diff
@@ -31,18 +34,28 @@ class Ppm2PbmControllerTest {
     @MockBean
     private ImageService imageService;
 
+    @Captor
+    private ArgumentCaptor<InputStream> inputStreamArgumentCaptor;
+
     @Test
     public void shouldReturnCorrectImage() throws Exception {
 
         String imageName = "3x3.ppm";
-        byte[] expectedBytes = Files.readAllBytes(Paths.get("src/test/resources/" + imageName));
-        String expectedContentType = "image/x-portable-graymap";
+        String expectedContentType = "image/png";
+
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + imageName)) {
+
+            byte[] expectedBytes = inputStream.readAllBytes();
+
+            given(imageService.getImage(imageService.getConvertedInputStream(any(InputStream.class)))).willReturn(new ByteArrayResource(expectedBytes));
 
-        given(imageService.getImage(anyString())).willReturn(new ByteArrayResource(expectedBytes));
+            this.mockMvc.perform(multipart("/convert")
+                            .file(new MockMultipartFile("file", imageName, "image/png", expectedBytes)))
+                    .andExpect(status().isOk())
+                    .andExpect(content().contentType(MediaType.parseMediaType(expectedContentType)))
+                    .andExpect(content().bytes(expectedBytes));
 
-        this.mockMvc.perform(get("/image/{imageName}", imageName))
-                .andExpect(status().isOk())
-                .andExpect(content().contentType(MediaType.parseMediaType(expectedContentType)))
-                .andExpect(content().bytes(expectedBytes));
+            verify(imageService).getImage(inputStreamArgumentCaptor.capture());
```

Comment:
Coś takiego, normalnie można by było zrobić jeszcze `getValue()` na tym captorze i na tej wartości asercję, ale że to input stream, to jest to obiekt jednorazowego użytku.

Alternatywnie można też po prostu 

`verify(imageService).gestImage(any(InputStream.class))` jeśli i tak z captora nic nie wyciągamy.